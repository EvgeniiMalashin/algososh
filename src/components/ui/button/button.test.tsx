import renderer from 'react-test-renderer';
import { Button } from './button';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Тестирование Button', () => {
  it('Button без текста', () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('Button с текстом', () => {
    const button = renderer.create(<Button text="test" />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('Button заблокированные', () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('Button c индикацией загрузки', () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('Button нажатие', () => {
    const callbackFunction = jest.fn();
    render(<Button text="testCallback" onClick={callbackFunction} />);
    const button = screen.getByText("testCallback");
    fireEvent.click(button);
    expect(callbackFunction).toHaveBeenCalledTimes(1);
  })
})