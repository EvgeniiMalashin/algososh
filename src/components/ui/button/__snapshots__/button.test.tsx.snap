// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование Button Button c индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование Button Button без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование Button Button заблокированные 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование Button Button с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;
